python/mxnet/metric.py

```diff
@@ -325,6 +371,15 @@ def reset(self):
         except AttributeError:
             pass
 
+    def reset_local(self):
+        """Resets the local portion of the internal evaluation results
+        to initial state."""
+        try:
+            for metric in self.metrics:
+                metric.reset_local()
+        except AttributeError:
+            pass
```

Comment:
This is not added in this PR and I'm not sure myself why this is needed here (this function is basically a copy of the `reset` function but calls `reset_local` on children instead of `reset`).